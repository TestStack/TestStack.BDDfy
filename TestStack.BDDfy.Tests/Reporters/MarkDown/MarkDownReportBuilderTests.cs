using System.Runtime.CompilerServices;
using ApprovalTests;
using NUnit.Framework;
using TestStack.BDDfy.Configuration;
using TestStack.BDDfy.Reporters;
using TestStack.BDDfy.Reporters.MarkDown;
using TestStack.BDDfy.Tests.Reporters.Html;

namespace TestStack.BDDfy.Tests.Reporters.MarkDown
{
    [TestFixture]
    public class MarkDownReportBuilderTests
    {
        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ShouldProduceExpectedMarkdown()
        {
            // somehow the scenario id keeps increasing on TC
            // resetting here explicitly
            Configurator.IdGenerator.Reset();

            // setting the culture to make sure the date is formatted the same on all machines
            using (new TemporaryCulture("en-GB"))
            {
                var model = new FileReportModel(new ReportTestData().CreateMixContainingEachTypeOfOutcome());

                var sut = new MarkDownReportBuilder();
                var result = sut.CreateReport(model);
                Approvals.Verify(result);
            }
        }
    }
}
