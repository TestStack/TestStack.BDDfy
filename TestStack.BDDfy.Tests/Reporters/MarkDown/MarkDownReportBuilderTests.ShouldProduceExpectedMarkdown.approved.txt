## Story: Account holder withdraws cash
 **As an account holder**  
 **I want to withdraw cash**  
 **So that I can get money when the bank is closed**  

### Happy Path Scenario
  Given a positive account balance  
  When the account holder requests money  
  Then money is dispensed  

### Inconclusive Scenario
  Given a negative account balance  
  When the account holder requests money  
  Then no money is dispensed  

### Not Implemented Scenario
  Given a negative account balance  
  When the account holder requests money  
  Then no money is dispensed  

### Sad Path Scenario
  Given a negative account balance  
  When the account holder requests money  
  Then no money is dispensed  

## Story: Happiness
 **As a person**  
 **I want ice cream**  
 **So that I can be happy**  

### Happy Path Scenario
  Given a positive account balance  
  When the account holder requests money  
  Then money is dispensed  

### Inconclusive Scenario
  Given a negative account balance  
  When the account holder requests money  
  Then no money is dispensed  

### Not Implemented Scenario
  Given a negative account balance  
  When the account holder requests money  
  Then no money is dispensed  

### Sad Path Scenario
  Given a negative account balance  
  When the account holder requests money  
  Then no money is dispensed  


### Happy Path Scenario
  Given a positive account balance  
  When the account holder requests money  
  Then money is dispensed  

### Inconclusive Scenario
  Given a negative account balance  
  When the account holder requests money  
  Then no money is dispensed  

### Not Implemented Scenario
  Given a negative account balance  
  When the account holder requests money  
  Then no money is dispensed  

### Sad Path Scenario
  Given a negative account balance  
  When the account holder requests money  
  Then no money is dispensed  

